roofit/batchcompute/src/RooBatchCompute.cu

```diff
@@ -0,0 +1,174 @@
+// RooBatchCompute library created September 2020 by Emmanouil Michalainas
+/**
+\file RooBatchCompute.cu
+\class RbcClass
+\ingroup Roobatchcompute
+
+This file contains the code for cuda computations using the RooBatchCompute library.
+**/ 
+#include "rbc.h"
+#include "Batches.h"
+
+#include "TError.h"
+
+#include <algorithm>
+#include <thrust/reduce.h>
+
+#ifndef RF_ARCH
+#error "RF_ARCH should always be defined"
+#endif
+
+namespace RooBatchCompute {
+namespace RF_ARCH {
+
+// For turning preprocessor definition into string
+#define _STRINGIFY(X) #X
+#define STRINGIFY(X) _STRINGIFY(X)
+  
+std::vector<void(*)(Batches)> getFunctions();
+/// This class overrides some RbcInterface functions, for the purpose of providing
+/// a cuda specific implementation of the library.
+class RbcClass : public RbcInterface {
+  private:
+    const std::vector<void(*)(Batches)> _computeFunctions;
```

Comment:
These should probably take `Batches` as `const &`?

Reply:
See https://github.com/root-project/root/pull/9004#discussion_r764703643